saleor/graphql/attribute/mutations/attribute_bulk_create.py

```diff
@@ -0,0 +1,525 @@
+                    ]
+                for value in values:
+                    value.full_clean(exclude=["attribute"])
+
+                instances_data_and_errors_list.append(
+                    {
+                        "instance": instance,
+                        "errors": index_error_map[index],
+                        "values": values,
+                    }
+                )
+            except ValidationError as exc:
+                for key, errors in exc.error_dict.items():
+                    for e in errors:
+                        index_error_map[index].append(
+                            AttributeBulkCreateError(
+                                path=to_camel_case(key),
+                                message=e.messages[0],
+                                code=e.code,
+                            )
+                        )
+                instances_data_and_errors_list.append(
+                    {"instance": None, "errors": index_error_map[index]}
+                )
+        return instances_data_and_errors_list
+
+    @classmethod
+    def save(cls, info, instances_data_with_errors_list):
+        attributes_to_create: list = []
+        values_to_create: list = []
+        for attribute_data in instances_data_with_errors_list:
+            attribute = attribute_data["instance"]
+
+            if not attribute:
+                continue
+
+            attributes_to_create.append(attribute)
+
+            if attribute_data["values"]:
+                values_to_create.extend(attribute_data["values"])
+
+        models.Attribute.objects.bulk_create(attributes_to_create)
+        models.AttributeValue.objects.bulk_create(values_to_create)
+
+        return attributes_to_create
+
+    @classmethod
+    def post_save_actions(cls, info, attributes):
+        manager = get_plugin_manager_promise(info.context).get()
+        for attribute in attributes:
+            cls.call_event(manager.attribute_created, attribute)
+
+    @classmethod
+    @traced_atomic_transaction()
+    def perform_mutation(cls, root, info, **data):
+        index_error_map: dict = defaultdict(list)
+        error_policy = data.get("error_policy", ErrorPolicyEnum.REJECT_EVERYTHING.value)
+
+        # clean and validate inputs
+        cleaned_inputs_map = cls.clean_attributes(
+            info, data["attributes"], index_error_map
+        )
+        instances_data_with_errors_list = cls.create_attributes(
+            info, cleaned_inputs_map, index_error_map
+        )
+
+        # check error policy
+        if any([True if error else False for error in index_error_map.values()]):
+            if error_policy == ErrorPolicyEnum.REJECT_EVERYTHING.value:
+                results = get_results(instances_data_with_errors_list, True)
+                return AttributeBulkCreate(count=0, results=results)
+
+            if error_policy == ErrorPolicyEnum.REJECT_FAILED_ROWS.value:
+                for data in instances_data_with_errors_list:
```

Comment:
Not sure if this clean up should be here. IMO `create_attributes` should set up `data["instance"]` as a None in case of the error. Or it should be done by `get_results`.  Now you have `create_attributes` that returns the data, for `reject_everything` the response is created in `get_results`, for `reject_failed_rows` the response data is initially prepared in `perform_mutation` and later it is converted in `get_results`